pkg/conduit/pipelines_parser_test.go

```diff
@@ -0,0 +1,151 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conduit
+
+import (
+	"bytes"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/conduitio/conduit/pkg/foundation/assert"
+)
+
+func Test_Parser(t *testing.T) {
+	filename, err := filepath.Abs("./test/pipelines1.yml")
+	if err != nil {
+		t.Error(err)
+	}
+
+	want := PipelinesInfo{
+		Version: "1.0",
+		Pipelines: map[string]PipelineInfo{
+			"pipeline1": {
+				Status: "running",
+				Config: PipelineConfig{"pipeline1", "desc1"},
+				Processors: ProcessorInfo{
+					"pipeline1proc1": {
+						Name: "pipeline processor",
+						Type: "transform",
+						Config: ProcessorConfig{
+							Settings: map[string]string{
+								"additionalProp1": "string",
+								"additionalProp2": "string",
+							},
+						},
+					},
+				},
+				Connectors: ConnectorInfo{
+					"con1": {
+						Type:   "source",
+						Plugin: "builtin:s3",
+						Config: ConnectorConfig{
+							Name: "s3-source",
+							Settings: map[string]string{
+								"aws.region": "us-east-1",
+								"aws.bucket": "my-bucket",
+							},
+						},
+						Processors: ProcessorInfo{
+							"proc1": {
+								Name: "connector processor",
+								Type: "transform",
+								Config: ProcessorConfig{
+									Settings: map[string]string{
+										"additionalProp1": "string",
+										"additionalProp2": "string",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"pipeline2": {
+				Status: "stopped",
+				Config: PipelineConfig{"pipeline2", "desc2"},
+				Connectors: ConnectorInfo{
+					"con2": {
+						Type:   "destination",
+						Plugin: "builtin:file",
+						Config: ConnectorConfig{
+							Name: "file-dest",
+							Settings: map[string]string{
+								"path": "my/path",
+							},
+						},
+						Processors: ProcessorInfo{
+							"con2proc1": {
+								Name: "connector processor",
+								Type: "transform",
+								Config: ProcessorConfig{
+									Settings: map[string]string{
+										"additionalProp1": "string",
+										"additionalProp2": "string",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"pipeline3": {
+				Status: "stopped",
+				Config: PipelineConfig{"pipeline3", "empty"},
+			},
+		},
+	}
+
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		t.Error(err)
+	}
+
+	got, err := Parse(bytes.NewReader(data))
+	assert.Nil(t, err)
+	assert.Equal(t, want, got)
+}
+
+func Test_Parser_Duplicate_Pipeline_Id(t *testing.T) {
+	filename, err := filepath.Abs("./test/pipelines2.yml")
+	if err != nil {
+		t.Error(err)
+	}
+
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		t.Error(err)
+	}
+
+	p, err := Parse(bytes.NewReader(data))
+	assert.Error(t, err)
```

Comment:
Ideally, whenever there's an error, we also make sure that the correct error is returned too. I was just looking at pipeline2.yml, and the version of the first doc is 1.1, so I have a feeling the error we have in the test here is not about a duplicated pipeline, but it's about an invalid version.